### PR TITLE
CORTX-31955 : Fixed codacy issue for redundant if condition and

### PR DESCRIPTION
**Pull request by papan-singh**
NULL check before use of pointer.

Signed-off-by: Papan Kumar Singh <papan.kumarsingh@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by papan-singh** [comment]
done unit testing
[root@ssc-vm-rhev4-2550 cortx-motr-1]# ./scripts/m0 run-ut -t cm-cp-ut
----- run_ut -t cm-cp-ut -----
START Iteration: 1 out of 1
cm-cp-ut
  cp-single_thread                                0.00 sec    0 B
  cp-multi_thread                                 0.01 sec    3 MiB
  [ time: 1.46 sec, mem: 84 MiB, leaked: 6 MiB ]

Time: 1.46 sec, Mem: 84 MiB, Leaked: 6 MiB, Asserts: 18
Unit tests status: SUCCESS
END   Iteration: 1 out of 1

utime 0.542909 stime 0.387714 maxrss 228312 nvcsw 3551 nivcsw 32
minflt 66548 majflt 7 inblock 264 oublock 33440
rchar 154741 wchar 1186 syscr 349 syscw 48
read_bytes 135168 write_bytes 17121280 cancelled_write_bytes 69632
[root@ssc-vm-rhev4-2550 cortx-motr-1]# ./scripts/m0 run-ut -t cm-ut
----- run_ut -t cm-ut -----
START Iteration: 1 out of 1
cm-ut
  cm_setup_ut                                     0.02 sec    1 MiB
  cm_setup_failure_ut                             0.00 sec   96 KiB
  cm_init_failure_ut                              0.00 sec    0 B
  cm_prepare_failure_ut                           0.00 sec  298 KiB
  cm_ready_failure_ut                             0.00 sec  812 KiB
  cm_start_failure_ut                             0.00 sec  975 KiB
  cm_ag_ut                                        0.00 sec  194 KiB
  [ time: 3.30 sec, mem: 185 MiB, leaked: 15 MiB ]

Time: 3.30 sec, Mem: 185 MiB, Leaked: 15 MiB, Asserts: 328
Unit tests status: SUCCESS
END   Iteration: 1 out of 1

utime 1.025433 stime 0.785367 maxrss 487468 nvcsw 5996 nivcsw 33
minflt 142392 majflt 6 inblock 128 oublock 300536
rchar 189909 wchar 1974 syscr 820 syscw 483
read_bytes 65536 write_bytes 153874432 cancelled_write_bytes 86016

**Comment by shailesh-vaidya** [comment]
retest this please

**Comment by cortx-admin** [comment]
<h1>Jenkins CI Result : <a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372/>Motr#1372</a></h1><h2>Motr Test Summary</h2><table><tr><th>Test Result</th><th>Count</th><th>Info</tr><tr><td>:x:Failed</td><td>18</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/01net target=_blank>01motr-single-node/01net</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/03console target=_blank>01motr-single-node/03console</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/02rpcping target=_blank>01motr-single-node/02rpcping</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/53clusterusage-alert target=_blank>02motr-single-node/53clusterusage-alert</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/61sns-repair-motr-1n-1f target=_blank>03motr-single-node/61sns-repair-motr-1n-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/72spiel-sns-motr-repair-quiesce target=_blank>03motr-single-node/72spiel-sns-motr-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/69sns-repair-motr-quiesce target=_blank>03motr-single-node/69sns-repair-motr-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/62sns-repair-motr-mf target=_blank>03motr-single-node/62sns-repair-motr-mf</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/70sns-failure-after-repair-quiesce target=_blank>03motr-single-node/70sns-failure-after-repair-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/63sns-repair-motr-1k-1f target=_blank>03motr-single-node/63sns-repair-motr-1k-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/60sns-repair-motr-1f target=_blank>03motr-single-node/60sns-repair-motr-1f</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/66sns-repair-motr-abort-quiesce target=_blank>03motr-single-node/66sns-repair-motr-abort-quiesce</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/64sns-repair-motr-ios-fail target=_blank>03motr-single-node/64sns-repair-motr-ios-fail</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/04sss target=_blank>03motr-single-node/04sss</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/65sns-repair-motr-abort target=_blank>03motr-single-node/65sns-repair-motr-abort</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/52motr-singlenode-sanity target=_blank>01motr-single-node/52motr-singlenode-sanity</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/71spiel-sns-motr-repair target=_blank>03motr-single-node/71spiel-sns-motr-repair</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/00userspace-tests target=_blank>01motr-single-node/00userspace-tests</a></p></details></td></tr><tr><td>:checkered_flag:Skipped</td><td>15</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/28sys-kvs target=_blank>01motr-single-node/28sys-kvs</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/35m0singlenode target=_blank>01motr-single-node/35m0singlenode</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/04initscripts target=_blank>01motr-single-node/04initscripts</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/37protocol target=_blank>01motr-single-node/37protocol</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/51kem target=_blank>02motr-single-node/51kem</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/20rpc-session-cancel target=_blank>02motr-single-node/20rpc-session-cancel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/10pver-assign target=_blank>02motr-single-node/10pver-assign</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/21fsync-single-node target=_blank>02motr-single-node/21fsync-single-node</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/13dgmode-io target=_blank>02motr-single-node/13dgmode-io</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/14poolmach target=_blank>02motr-single-node/14poolmach</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/11m0t1fs target=_blank>02motr-single-node/11m0t1fs</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/26motr-user-kernel-tests target=_blank>02motr-single-node/26motr-user-kernel-tests</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/08spiel target=_blank>02motr-single-node/08spiel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/06conf target=_blank>03motr-single-node/06conf</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/36spare-reservation target=_blank>03motr-single-node/36spare-reservation</a></p></details></td></tr><tr><td>:heavy_check_mark:Passed</td><td>11</td><td><details><summary>:file_folder:</summary><p><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/43m0crate target=_blank>01motr-single-node/43m0crate</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/05confgen target=_blank>01motr-single-node/05confgen</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/06hagen target=_blank>01motr-single-node/06hagen</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/01motr-single-node/01kernel-tests target=_blank>01motr-single-node/01kernel-tests</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/07m0d-fatal target=_blank>02motr-single-node/07m0d-fatal</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/67fdmi-plugin-multi-filters target=_blank>02motr-single-node/67fdmi-plugin-multi-filters</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/02motr-single-node/41motr-conf-update target=_blank>02motr-single-node/41motr-conf-update</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/08spiel-multi-confd target=_blank>03motr-single-node/08spiel-multi-confd</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/24motr-dix-repair-lookup-insert-spiel target=_blank>03motr-single-node/24motr-dix-repair-lookup-insert-spiel</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/68sns-repair-motr-shutdown target=_blank>03motr-single-node/68sns-repair-motr-shutdown</a></br><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit/03motr-single-node/24motr-dix-repair-lookup-insert-m0repair target=_blank>03motr-single-node/24motr-dix-repair-lookup-insert-m0repair</a></p></details></td></tr><tr><td>Total</td><td>44</td><td><a href=https://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/1372//testReport/junit>:link:</a></td></tr></table><h2> CppCheck Summary</h2><p> &emsp;&emsp;&emsp;Cppcheck: No new warnings found :thumbsup:</p>

**Review by nkommuri — APPROVED** [review]

File: cm/ag.c [hunk 1]
```diff
@@ -345,7 +345,7 @@ M0_INTERNAL int m0_cm_aggr_group_alloc(struct m0_cm *cm,
 	M0_ASSERT(rc <= 0);
 	if (rc == 0 || rc == -ENOBUFS)
 		m0_cm_aggr_group_add(cm, *out, has_incoming);
-	else if (rc != 0)
+	else
```
[the remark this answers is not in the document]
Comment (papan-singh) [reply cm/ag.c:348]: we can remove these lines

[the remark this answers is not in the document]
Comment (papan-singh) [reply cm/ag.c]: incorporated

**Review by mehjoshi — APPROVED** [review]

**Pull request merged**